packages/next/src/server/use-cache/use-cache-wrapper.ts

```diff
@@ -260,6 +280,15 @@ async function collectResult(
   }
   // Propagate tags/revalidate to the parent context.
   propagateCacheLifeAndTags(outerWorkUnitStore, entry)
+
+  const cacheSignal =
+    outerWorkUnitStore && outerWorkUnitStore.type === 'prerender'
+      ? outerWorkUnitStore.cacheSignal
+      : null
+  if (cacheSignal) {
+    cacheSignal.endRead()
```

Comment:
Ideally begin and end should be more colocated but because `"use cache"` returns early while streaming, I to shove this in here.